go/vt/vtgate/planbuilder/operators/update.go

```diff
@@ -125,6 +125,12 @@ func createOperatorFromUpdate(ctx *plancontext.PlanningContext, updStmt *sqlpars
 		return nil, vterrors.VT12001("update with limit with foreign key constraints")
 	}
 
+	// Now we check if any of the foreign key columns that are being udpated have dependencies on other updated columns.
+	// This is unsafe, and we currently don't support this in Vitess.
+	if ctx.SemTable.IsFkDependentColumnUpdated(updStmt.Exprs) {
+		return nil, vterrors.VT12001("same column referenced in foreign key column update is also updated")
+	}
```

Comment:
better to give more details about the columns involved which resulted in this error.

Reply:
Done